compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5403,4 +5414,28 @@ private STNode parseLocalTypeDefinitionStatement(STNode annots) {
             typeDescriptor,
             semicolon);
     }
+
+    /**
+     * Parse nil literal.
+     *
+     * @return Parsed node
+     */
+    private STNode parseNilLiteral() {
+        STToken token = peek();
+        switch (token.kind) {
+            case NULL_KEYWORD:
+                STNode nullKeyword = consume();
+                STNode emptyNode = STNodeFactory.createEmptyNode();
+                return STNodeFactory.createNilLiteral(nullKeyword, emptyNode);
+            case OPEN_PAREN_TOKEN:
+                startContext(ParserRuleContext.NIL_LITERAL);
+                STNode openParenthesisToken = parseOpenParenthesis();
+                STNode closeParenthesisToken = parseCloseParenthesis();
+                endContext();
+                return STNodeFactory.createNilLiteral(openParenthesisToken, closeParenthesisToken);
+            default:
+                Solution sol = recover(token, ParserRuleContext.NIL_LITERAL);
+                return sol.recoveredNode;
+        }
```

Comment:
Is this approach ok?

Reply:
@SupunS No switch case means, `( )` is considered to be `NIL_LITERAL` kind and `NULL` is considered to be `NULL_KEYWORD` kind, and will have separate `pareseNilLiteral( )`, `pareseNullKeyword( )` methods ?

Reply:
yes. For the null keyword, we can do the same way as, string literal